src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+
+        public Locations( final String evidenceDir, final String intervalFile,
+                          final String qNamesMappedFile, final String kmerFile, final String qNamesAssemblyFile ) {
+            this.evidenceDir = evidenceDir;
+            this.intervalFile = intervalFile;
+            this.qNamesMappedFile = qNamesMappedFile;
+            this.kmerFile = kmerFile;
+            this.qNamesAssemblyFile = qNamesAssemblyFile;
+        }
+    }
+
+    /** helper class for calculating mean template length */
+    @DefaultSerializer(ReadCountAndLength.Serializer.class)
+    private static final class ReadCountAndLength {
+        private final long count;
+        private final long length;
+
+        ReadCountAndLength( final GATKRead read ) {
+            this.count = 1;
+            this.length = (read.isPaired() ? 2 : 1)*read.getLength();
+        }
+
+        ReadCountAndLength( final ReadCountAndLength countNLength1, final ReadCountAndLength countNLength2 ) {
+            this.count = countNLength1.count + countNLength2.count;
+            this.length = countNLength1.length + countNLength2.length;
+        }
+
+        private ReadCountAndLength( final Kryo kryo, final Input input ) {
+            count = input.readLong();
+            length = input.readLong();
+        }
+
+        private void serialize( final Kryo kryo, final Output output ) {
+            output.writeLong(count);
+            output.writeLong(length);
+        }
+
+        public int getMeanLength() { return count != 0 ? (int)(length/count) : 0; }
+
+        public static final class Serializer extends com.esotericsoftware.kryo.Serializer<ReadCountAndLength> {
+            @Override
+            public void write( final Kryo kryo, final Output output, final ReadCountAndLength readCountAndLength ) {
+                readCountAndLength.serialize(kryo, output);
+            }
+
+            @Override
+            public ReadCountAndLength read( final Kryo kryo, final Input input,
+                                            final Class<ReadCountAndLength> klass ) {
+                return new ReadCountAndLength(kryo, input);
+            }
+        }
+    }
+
+    /**
+     * A class that acts as a filter for breakpoint evidence.
+     * It passes only that evidence that is part of a putative cluster.
+     */
+    private static final class BreakpointClusterer
+            implements Function<BreakpointEvidence, Iterator<BreakpointEvidence>> {
+        private final int staleEventDistance;
+        private final SortedMap<BreakpointEvidence, Boolean> locMap = new TreeMap<>();
+        private final List<Map.Entry<BreakpointEvidence, Boolean>> reportableEntries = new ArrayList<>(2*MIN_EVIDENCE);
+        private final Iterator<BreakpointEvidence> noEvidence = Collections.emptyIterator();
+        private int currentContig = -1;
+
+        private static final int MIN_EVIDENCE = 15; // minimum evidence count in a cluster
```

Comment:
This seems like something we might want to play with so it'd be nice to turn this into a command-line parameter.